design.md

```diff
@@ -0,0 +1,107 @@
+## Design Document - v0
+
+This is an oversimplified version of a kademlia/mainline DHT.
+
+For at least the v0 of our DHT we only care about storing and retrieving information about other DHT clients/servers; which we will call Peers.
+
+Mainline separates peers from Peers as one is the DHT client/server and the other is a BitTorrent provider.  
+We do not care about anything more than the DHT clients/servers for now, so this should simplify things a bit.
+
+## Terminology
+
+* `peer` - A participating client/server in the DHT network.
+* `k-buckets` - The routing table keeps a list of known GOOD peers.
+* `alpha` - A small number representing the degree of parallelism in network calls, usually 3.
```

Comment:
> In summary, these observations show that strict parallelism
> with α = 3 is a good choice for the current Kad network.
> Higher values of α and loose parallelism substantially increase
> overhead without much change in performance. Also, Figure 5
> provides strong evidence for the correctness of our analysis in
> Section III.

According to [Improving Lookup Performance over a Widely-Deployed DHT](http://www.barsoom.org/papers/infocom-2006-kad.pdf)